test/src/main/java/org/apache/accumulo/test/CorruptMutationIT.java

```diff
@@ -0,0 +1,122 @@
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.accumulo.core.client.Accumulo;
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.clientImpl.ClientContext;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.dataImpl.KeyExtent;
+import org.apache.accumulo.core.dataImpl.thrift.TMutation;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata;
+import org.apache.accumulo.core.rpc.ThriftUtil;
+import org.apache.accumulo.core.rpc.clients.ThriftClientTypes;
+import org.apache.accumulo.core.tabletserver.thrift.TDurability;
+import org.apache.accumulo.core.tabletserver.thrift.TabletClientService;
+import org.apache.accumulo.core.trace.TraceUtil;
+import org.apache.accumulo.core.trace.thrift.TInfo;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.thrift.TApplicationException;
+import org.apache.thrift.TServiceClient;
+import org.junit.jupiter.api.Test;
+
+public class CorruptMutationIT extends AccumuloClusterHarness {
+  @Test
+  public void testCorruptMutation() throws Exception {
+
+    String table = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      c.tableOperations().create(table);
+      try (BatchWriter writer = c.createBatchWriter(table)) {
+        Mutation m = new Mutation("1");
+        m.put("f1", "q1", new Value("v1"));
+        writer.addMutation(m);
+      }
+
+      var ctx = (ClientContext) c;
+      var tableId = ctx.getTableId(table);
+      var extent = new KeyExtent(tableId, null, null);
+      var tabletMetadata = ctx.getAmple().readTablet(extent, TabletMetadata.ColumnType.LOCATION);
+      var location = tabletMetadata.getLocation();
+      assertNotNull(location);
+      assertEquals(TabletMetadata.LocationType.CURRENT, location.getType());
+
+      TabletClientService.Iface client =
+          ThriftUtil.getClient(ThriftClientTypes.TABLET_SERVER, location.getHostAndPort(), ctx);
+      // Make the same RPC calls made by the BatchWriter, but pass a corrupt serialized mutation in
+      // this try block.
+      try {
+        TInfo tinfo = TraceUtil.traceInfo();
+
+        long sessionId = client.startUpdate(tinfo, ctx.rpcCreds(), TDurability.DEFAULT);
+        Mutation m = new Mutation("abc");
+        m.put("x", "y", "z");
+
+        // Serialize the mutation
+        TMutation tMutation = m.toThrift();
+
+        // Simulate data corruption in the serialized mutation
+        tMutation.entries = -42;
+
+        // The server side will see an error here, however since this is a thrift oneway method no
+        // exception is expected here.
+        client.applyUpdates(tinfo, sessionId, extent.toThrift(), List.of(tMutation));
+
```

Comment:
You might want to try and insert a valid mutation before and after the corrupt one. The one before should be applied, but the one after should be ignored, right?

Reply:
That is a good change to the test, added in 365c36b.  The servers tries to deserialize all of the mutations in the list first thing and when one fails it tosses the whole thing before doing anything else w/ the mutations.

Reply:
> I was thinking of multiple applyUpdates calls when I made the suggestion, another one before and after the corrupt one, not necessarily in the same batch. Do you think that should be added as well, or do you think we have coverage of that already?

oh, ok.  That would be good to test too, however its behavior would be indeterminate.  The tserver buffers data and eventually flushes a batch of data through the write process, however it does not do this for each applyUpdate RPC call.  It just does based on the amount of data that has built up from one or more apply update RPC calls.    Could have the test have a set of data is must see and set of data it may or may not see.

Reply:
If you set `TSERV_TOTAL_MUTATION_QUEUE_MAX` to something stupidly low like 10 bytes, would that work?

Reply:
> If you set TSERV_TOTAL_MUTATION_QUEUE_MAX to something stupidly low like 10 bytes, would that work?

oh yeah that might work.  Let me try that and see what happens.  Made an update in 4f83927 to make multuple applyUpdate calls.

Reply:
Setting TSERV_TOTAL_MUTATION_QUEUE_MAX worked nicely.  It seemed like it was not working after first, but there was a bug in the test code.  Updated in 3e7a79e.  Opened #5235 after looking into why it did not seem to be working, noticed that issue.  I need to make some changes for #5235 in this PR for the error case, but that does not handle the client dying case.

Reply:
Pushed 0f93150 which partially addresses #5235.  I manually added and remove some logging to test the change and ran the new test in this PR.  Not sure how to test that generally though.